osf/models/base.py

```diff
@@ -304,8 +304,8 @@ def load(cls, q, select_for_update=False):
         try:
             # guids___id__isnull=False forces an INNER JOIN
             if select_for_update:
-                return cls.objects.filter(guids___id__isnull=False, guids___id=q).select_for_update().first()
-            return cls.objects.filter(guids___id__isnull=False, guids___id=q).first()
+                return cls.objects.filter(guids___id__isnull=False, guids___id=q).select_for_update()[:1].get()
+            return cls.objects.filter(guids___id__isnull=False, guids___id=q)[:1].get()
```

Comment:
Curiosity question, what's the advantage of `[:1].get()` vs. `.first()`?

Reply:
This query was showing up as one of the slow ones, and I read somewhere that `first()` is sometimes slower than `[:1].get()`, because it does an `order_by`. When I profiled with 10 trials apiece. I didn't think the order_by was necessary, but let me know if I'm wrong.


  | STDEV(ms) | AVG(ms)
-- | -- | --
Node.objects.filter(guids___id__isnull=False, guids___id=q).first() | 0.001571882113 | 0.0050574
Node.objects.filter(guids___id__isnull=False, guids___id=q)[:1].get() | 0.0008734698685 | 0.0028995

Reply:
+1. `.first()` does an unnecessary `ORDER BY`.

Turns out `[0]` is the same as `[:1].get()`.

```python


In [1]: Node.objects.filter()[:1].get()
SELECT "osf_abstractnode"."id",
       ...
FROM "osf_abstractnode"
WHERE "osf_abstractnode"."type" = 'osf.node'
LIMIT 1


In [2]: Node.objects.filter()[0]
SELECT "osf_abstractnode"."id",
       ...
FROM "osf_abstractnode"
WHERE "osf_abstractnode"."type" = 'osf.node'
LIMIT 1

In [3]: Node.objects.filter().first()
SELECT "osf_abstractnode"."id",
       ...
FROM "osf_abstractnode"
WHERE "osf_abstractnode"."type" = 'osf.node'
ORDER BY "osf_abstractnode"."id" ASC
LIMIT 1
```

What you have is good as is.

Reply:
Ohh, neat. Thanks for the explanation @sloria and @pattisdr!